### ISSUE TITLE
The one jar is not updated when a dependant sup-projects jar is updated

### ISSUE THREAD

**Issue by walec51** [rholder/gradle-one-jar#12]

**Comment by walec51** [comment]
I'll give more info tomorrow

**Comment by walec51** [comment]
I have a multi project configuration. My client which is onejared depends directly on a jar library subproject.
If I modify the jar library subproject the onejar will not get the new jar library. I've added a workaround:

task oneJar(type: OneJar) {
    mainClass = "com.prodoko.base.client.example.ExampleClientAppLauncher"
    outputs.upToDateWhen { false }
}

and it works for the direct dependency. But if I modify other jar library sub-projects on which my onejar app depends indirectly the wont get included even with the workaround.

My whole project is available at:
https://bitbucket.org/walczak_it/prodoko-base/

prodoko-base-client-example - is the onejared app

**Comment by rholder** [comment]
Thanks for the detailed explanation, and the possible workaround that you've tried. I'll try to block out some time this weekend to take a look at your project to see if I can reproduce the behavior. Can you tell me what version of Gradle you've been using just in case something changed since the plugin was released?

**Comment by walec51** [comment]
In the console I've used:

Gradle 1.6

Gradle build time: wtorek, 7 maja 2013 09:12:14 UTC
Groovy: 1.8.6
Ant: Apache Ant(TM) version 1.8.4 compiled on May 22 2012
Ivy: 2.2.0
JVM: 1.8.0-ea (Oracle Corporation 25.0-b42)
OS: Linux 3.8.0-27-generic amd64

and also the build in gradle in eclipse 4.2.

oh, my project uses snapshot builds of jdk 8

but I can create a mini test project in jdk 7 to isolate this problem if you'd like.

**Comment by rholder** [comment]
Thanks for the version info. I've actually run into a similar behavior before with code not getting updated on a couple of multi-module projects with Gradle (even those that didn't include the `gradle-one-jar` plugin). If I remember correctly, dropping this snippet into the parent module seemed to solve my issues with always keeping things up to date when developing on a project before release:

``` groovy
subprojects {
    configurations {
        all {
            resolutionStrategy {
                cacheChangingModulesFor 0, 'seconds'
            }
        }
    }
}
```

I wonder if using a concrete release version number of `0.1` in your `gradle.properties` (instead of something like `0.1-SNAPSHOT`) might be causing Gradle not to realize that the dependent local modules are `changing = true`. You might check [here](http://www.gradle.org/docs/current/userguide/dependency_management.html#sub:dynamic_versions_and_changing_modules) for the related official docs on how those are cached (which are still a bit fuzzy, in my opinion).

I also noticed a possible `AntBuilder` bug in the `gradle-one-jar` code as well that could be contributing to Gradle not correctly calculating the dependency chain for output files on multi-module projects. I'm currently testing it and will update this thread if it does or doesn't seem to fix things.

**Comment by rholder** [comment]
I'm going to mark this as closed in the latest version of Gradle, 1.11. If it's still an issue for you or anyone else, please feel free to reopen.
